cli/azd/pkg/project/service.go

```diff
@@ -95,3 +96,23 @@ func (svc *Service) Deploy(
 
 	return result, progress
 }
+
+func (svc *Service) getOverriddenEndpoints() []string {
+	overriddenEndpoints := svc.Environment.GetServiceProperty(svc.Config.Name, "ENDPOINTS")
+	if overriddenEndpoints != "" {
+		var endpoints []string
+		err := json.Unmarshal([]byte(overriddenEndpoints), &endpoints)
+		if err != nil {
+			// This can only happen if the environment file is manually edited.
+			// For typical infra provider output passthrough, the infra provider would guarantee well-formed syntax
+			log.Printf(
+				"failed to unmarshal endpoints override for service '%s' as JSON: %v, skipping override",
```

Comment:
If you end up having to do another iteration, maybe say "a JSON array of strings"? (Although I suspect the error we include from the unmarshaller might give us this hint as well)